src/screens/ui/ActivityAutoCompletionScreen.tsx

```diff
@@ -0,0 +1,170 @@
+  return (
+    <ImageBackground>
+      <YStack flex={1} px={20} gap={20}>
+        {children}
+      </YStack>
+    </ImageBackground>
+  );
+};
+
+const TimesUpScreen: FC<Props> = ({ onPressDone }) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <FlexContainer justifyContent="flex-end">
+        <TimerSandIcon color="#000" size={80} />
+        <Text fontSize={25}>{t('autocompletion:time_limit_reached')}</Text>
+      </FlexContainer>
+      <FlexContainer justifyContent="flex-start">
+        <Text textAlign="center">
+          {t('autocompletion:time_expired', {
+            activityNames: '[First Activity, Second Activity]',
+          })}
+        </Text>
+
+        <Box w="100%">
+          <SubmitButton
+            accessibilityLabel="autocompletion-times-up"
+            mode="dark"
+            onPress={onPressDone}
+            borderRadius={20}
+          >
+            {t('autocompletion:done')}
+          </SubmitButton>
+        </Box>
+      </FlexContainer>
+    </>
+  );
+};
+
+const ProcessingAnswersScreen: FC = () => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <FlexContainer justifyContent="flex-end">
+        <Spinner size={80} />
+        <Text fontSize={25}>{t('autocompletion:processing_answers')}</Text>
+      </FlexContainer>
+
+      <FlexContainer justifyContent="flex-start">
+        <Text textAlign="center">{t('autocompletion:preparing_answers')}</Text>
+      </FlexContainer>
+    </>
+  );
+};
+
+const AnswersSubmittedScreen: FC<Props> = ({ onPressDone }) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <FlexContainer justifyContent="flex-end">
+        <Box bg="$alertSuccessIcon" borderRadius={100} p={10}>
+          <CheckIcon color="white" size={60} />
+        </Box>
+        <Text fontSize={25}>{t('autocompletion:answers_submitted')}</Text>
+      </FlexContainer>
+      <FlexContainer justifyContent="flex-start">
+        <Box w="100%">
+          <SubmitButton
+            accessibilityLabel="autocomplete-answers-submitted"
+            mode="dark"
+            onPress={onPressDone}
+            borderRadius={20}
+          >
+            {t('autocompletion:done')}
+          </SubmitButton>
+        </Box>
+      </FlexContainer>
+    </>
+  );
+};
+
+const ActivityAutoCompletionScreen: FC = () => {
+  const {
+    isCompleted,
+    isLoading,
+    process: processQueue,
+  } = useQueueProcessing();
+
+  const { isAlertOpened: isRetryAlertOpen, openAlert: openRetryAlert } =
+    useRetryUpload({
+      retryUpload: processQueue,
+    });
+
+  const { process: processWithAutocompletion } = useAutoCompletion();
+  const { goBack } = useNavigation();
+
+  const processAnswers = async () => {
+    const success = await processWithAutocompletion();
+
+    if (!success) {
+      openRetryAlert();
+    }
+  };
+
+  if (isRetryAlertOpen) {
+    return <MainContainer />;
+  }
+
+  if (isCompleted) {
```

Comment:
"reset" state is not ensured. Check UploadObservable, call reset() as it's in other places.